#! /usr/bin/env python

PACKAGE='omni_vision'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#  Name   Type    Reconfiguration level  Description   Default    Min   Max
gen.add("show",         bool_t,  0,   "print screen",          False)
gen.add("ball",         bool_t,  0,   "show the ball",         True)
gen.add("white",        bool_t,  0,   "show the white_points", True)
gen.add("obstacle",     bool_t,  0,   "show the obstacles",    True)
gen.add("scan",         bool_t,  0,   "show the scanpoints",   False)
gen.add("obsthres",      int_t,  0,   "obsthres", 50, 30, 150)
gen.add("obs_length_thres",    double_t,  0,   "obstacles_length_thres",   0.30,  0.01, 1)
gen.add("obs_basic_thres",     double_t,  0,   "obstacles_basic_thres",    0.05, 0.01, 1)

exit(gen.generate(PACKAGE, "omni_vision", "OmniVision"))
